igvc_navigation/src/trajectory_planner/motion_profiler.cpp

```diff
@@ -0,0 +1,86 @@
+#include "motion_profiler.h"
+#include <tf/transform_datatypes.h>
+
+MotionProfiler::MotionProfiler(double axle_length, const WheelConstraint& wheel_constraint,
+                               const RobotConstraint& robot_constraint,
+                               const MotionProfilerOptions& motion_profiler_options, double target_velocity)
+  : wheel_constraint_{ wheel_constraint }
+  , robot_constraint_{ robot_constraint }
+  , motion_profiler_options_{ motion_profiler_options }
+  , axle_length_{ axle_length }
+  , target_velocity_{ target_velocity }
+{
+}
+
+void MotionProfiler::profileTrajectory(const igvc_msgs::trajectoryPtr& trajectory_ptr, const RobotState& state)
+{
+  // For now, a simple "motion profile" where velocity is a function of curvature. ie. as curvature increases,
+  // velocity decreases.
```

Comment:
You're also profiling based on acceleration.

Reply:
that's true.